cg/meta/upload/gt.py

```diff
@@ -39,97 +43,139 @@ def data(self, analysis: Analysis) -> dict:
+        """Parse and return a QC metrics file."""
         qcmetrics_raw: dict = ReadFile.get_content_from_file(
-            file_format=FileFormat.YAML, file_path=qc_metrics
+            file_format=FileFormat.YAML, file_path=qc_metrics_file
         )
         return MIPMetricsDeliverables(**qcmetrics_raw)
 
-    def upload(self, data: dict, replace: bool = False):
-        """Upload data about genotypes for a family of samples."""
-        self.gt.upload(str(data["bcf"]), data["samples_sex"], force=replace)
-
-    @staticmethod
-    def _is_variant_file(genotype_file: File):
-        return genotype_file.full_path.endswith("vcf.gz") or genotype_file.full_path.endswith("bcf")
-
-    def _get_genotype_files(self, version_id: int) -> list:
-        return self.hk.files(version=version_id, tags=["genotype"]).all()
+    def get_samples_sex_raredisease(
+        self, case: Case, hk_version: Version
+    ) -> dict[str, dict[str, str]]:
+        """Return sex information from StatusDB and from analysis prediction (stored Housekeeper QC metrics file)."""
+        qc_metrics_file: Path = self.get_qcmetrics_file(hk_version)
+        samples_sex: dict[str, dict[str, str]] = {}
+        for case_sample in case.links:
+            sample_id: str = case_sample.sample.internal_id
+            samples_sex[sample_id] = {
+                "pedigree": case_sample.sample.sex,
+                "analysis": self.get_analysis_sex_raredisease(qc_metrics_file, sample_id),
+            }
+        return samples_sex
 
-    @staticmethod
-    def is_suitable_for_genotype_upload(case_obj: Case) -> bool:
-        """Check if a cancer case is contains WGS and normal sample."""
+    def get_analysis_sex_raredisease(self, qc_metrics_file: Path, sample_id: Sample) -> str:
+        """Return analysis sex for each sample of an analysis."""
+        qc_metrics: list[MetricsBase] = self.get_parsed_qc_metrics_data_raredisease(qc_metrics_file)
+        return str(
+            next(
+                metric.value
+                for metric in qc_metrics
+                if metric.name == RAREDISEASE_PREDICTED_SEX_METRIC and metric.id == sample_id
+            )
+        )
 
-        samples: list[Sample] = case_obj.samples
-        return any(
-            (not sample.is_tumour and PrepCategory.WHOLE_GENOME_SEQUENCING == sample.prep_category)
-            for sample in samples
+    def get_parsed_qc_metrics_data_raredisease(self, qc_metrics: Path) -> list[MetricsBase]:
+        """Parse and return a QC metrics file."""
+        qcmetrics_raw: dict = ReadFile.get_content_from_file(
+            file_format=FileFormat.YAML, file_path=qc_metrics
         )
+        return [MetricsBase(**metric) for metric in qcmetrics_raw["metrics"]]
+
+    def get_qcmetrics_file(self, hk_version_obj: Version) -> Path:
+        """Return a QC metrics file path."""
+        hk_qcmetrics: Path = self.hk.files(
+            version=hk_version_obj.id, tags=HkAnalysisMetricsTag.QC_METRICS
```

Comment:
It accepts a set of metrics

```suggestion
            version=hk_version_obj.id, tags={HkAnalysisMetricsTag.QC_METRICS}
```